trieste/models/gpflow/architectures.py

```diff
@@ -0,0 +1,159 @@
+    appropriate. Using priors allows for using maximum a posteriori estimate of these kernel
+    parameters during model fitting.
+
+    :param data: Dataset from the initial design, used for estimating the variance of observations.
+    :param trainable_likelihood: If set to `True` likelihood paramater is set to
+        trainable. By default set to `False`.
+    :param priors: If set to `True` (default) priors are set for kernel parameters.
+    :return: A :class:`~gpflow.models.GPR` model.
+    """
+    empirical_variance = tf.math.reduce_variance(data.observations)
+    dim = data.query_points.shape[-1]
+
+    kernel = gpflow.kernels.Matern52(variance=empirical_variance, lengthscales=[0.2] * dim)
+
+    if kernel_priors:
+        prior_scale = tf.cast(1.0, dtype=gpflow.default_float())
+        kernel.variance.prior = tfp.distributions.LogNormal(
+            tf.math.log(empirical_variance), prior_scale
+        )
+        kernel.lengthscales.prior = tfp.distributions.LogNormal(
+            tf.math.log(kernel.lengthscales), prior_scale
+        )
+
+    model = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+
+    if not trainable_likelihood:
+        gpflow.set_trainable(model.likelihood, False)
+
+    return model
+
+
+def build_sgpr(
+    data: Dataset,
+    search_space: Optional[Box] = None,
+    trainable_likelihood: bool = False,
+    trainable_inducing_points: bool = False,
+) -> SGPR:
+    empirical_variance = tf.math.reduce_variance(data.observations)
+    dim = data.query_points.shape[-1]
+    num_data = len(data.observations)
+
+    kernel = gpflow.kernels.Matern52(variance=empirical_variance, lengthscales=[0.2] * dim)
+
+    if search_space:
+        inducing_points = search_space.sample_sobol(num_data)
+    else:
+        inducing_points = data.query_points[:num_data, :]
+
+    model = gpflow.models.SGPR(data.astuple(), kernel, inducing_points, noise_variance=1e-5)
+
+    if not trainable_likelihood:
+        gpflow.set_trainable(model.likelihood, False)
+
+    if not trainable_inducing_points:
+        gpflow.set_trainable(model.inducing_variable, False)
+
+    return model
+
+
+def build_vgp(data: Dataset, trainable_likelihood: bool = False) -> VGP:
+    empirical_variance = tf.math.reduce_variance(data.observations)
+    dim = data.query_points.shape[-1]
+
+    kernel = gpflow.kernels.Matern52(variance=empirical_variance, lengthscales=[0.2] * dim)
+
+    likelihood = gpflow.likelihoods.Gaussian(variance=1e-5)
+    model = gpflow.models.VGP(data.astuple(), kernel, likelihood)
+
+    if not trainable_likelihood:
+        gpflow.set_trainable(model.likelihood, False)
+
+    return model
+
+
+def build_vgp_classification(data: Dataset, trainable_likelihood: bool = False) -> VGP:
+    empirical_variance = tf.math.reduce_variance(data.observations)
+    dim = data.query_points.shape[-1]
+
+    kernel = gpflow.kernels.Matern52(variance=empirical_variance, lengthscales=[0.2] * dim)
```

Comment:
same as SGPR: priors, lengthscales

Reply:
I don't think that the empirical variance makes sense here. The variance encodes the sharpness of the transition from one class to the other. We should just use 1. as default, unless we know that the data is deterministic, in that case we should use a larger non-trained value (cf failure notebook)

Reply:
failure notebook also sets kernel variance as nontrainable - this should be for both the determinisitc scenario and noisy scenario, right?

Reply:
Deterministic scenario: variance is set to a large, non-trainable value (e.g. 10)

Noisy scenario: variance is trainable, we can initialise it at a neutral value, e.g. 1.0